src/Compilers/CSharp/Portable/Binder/Binder_Deconstruct.cs

```diff
@@ -568,7 +568,7 @@ private static void FlattenDeconstructVariables(ArrayBuilder<DeconstructionVaria
                 result.WasCompilerGenerated = true;
                 diagnostics.AddRange(bag);
 
-                if (bag.HasAnyErrors())
+                if (bag.HasAnyResolvedErrors())
                 {
```

Comment:
> This code path doesn't occur in member declaration processing, and so the resolution status can be ignored.

It is likely that some expressions are bound during declaration processing. Therefore, it is much safer to never trigger resolution of lazy diagnostics during binding.